build_tools/update_translations.fish

```diff
@@ -71,10 +78,23 @@ if set -l --query extract
     or exit 1
 end
 
+if set -l --query _flag_dry_run
+    # On a dry run, we do not modify po/ but write to a temporary directory instead and check if
+    # there is a difference between po/ and the tmpdir after re-generating the PO files.
+    set -g tmpdir (mktemp -d)
+    # On a dry-run, we do not update the MO files.
+    set -l --erase mo
+end
+
 for po_file in $po_files
+    if set -g --query tmpdir
+        cp $po_file $tmpdir
+        set -f po_file $tmpdir/(basename $po_file)
+    end
     if set -l --query po
         if test -e $po_file
-            msgmerge --update --no-fuzzy-matching --no-wrap --backup=none $po_file $template_file
+            msgmerge --update --no-fuzzy-matching --no-wrap --backup=none --quiet $po_file $template_file
+            or set exit_status $status
```

Comment:
should also  declare this variable (global/local) to prevent weird errors on env injection